docs/source/tutorials_source/package/tutorial_active_learning.py

```diff
@@ -0,0 +1,177 @@
+host = "https://api.lightly.ai"  # the URL of the web platform
+token = "ABCDEF"  # the token of the web platform
+dataset_id = "ASDFASDF"  # the id of your dataset on the web platform
+n_samples_to_sample = [100, 200]  # We first want to 100 samples to be labeled and
+# then another 100 to have 200 samples in total
+methods_to_sample = [SamplingMethod.CORESET, SamplingMethod.CORAL]  # We perform the initial sampling with CORESET
+# and then use CORAL for the following sampling with active learning scores
+
+
+# %%
+# Definition of a dataset for the classifier based on the embeddings.csv
+# The kNN-classifier needs the embeddings as features for its classification. Thus we define a class to create
+# such a dataset out of the embeddings.csv. It also allows to choose only a subset of all samples
+# dependent on the filenames given.
+
+class CSVEmbeddingDataset:
+    def __init__(self, path_to_embeddings_csv: str):
+        with open(path_to_embeddings_csv, 'r') as f:
+            data = csv.reader(f)
+
+            rows = list(data)
+            header_row = rows[0]
+            rows_without_header = rows[1:]
+
+            index_filenames = header_row.index('filenames')
+            filenames = [row[index_filenames] for row in rows_without_header]
+
+            index_labels = header_row.index('labels')
+            labels = [row[index_labels] for row in rows_without_header]
+
+            embeddings = rows_without_header
+            indexes_to_delete = sorted([index_filenames, index_labels], reverse=True)
+            for embedding_row in embeddings:
+                for index_to_delete in indexes_to_delete:
+                    del embedding_row[index_to_delete]
+
+        self.dataset = dict([(filename, (np.array(embedding_row, dtype=float), int(label)))
+                             for filename, embedding_row, label in zip(filenames, embeddings, labels)])
+
+    def get_features(self, filenames: List[str]) -> np.ndarray:
+        features_array = np.array([self.dataset[filename][0] for filename in filenames])
+        return features_array
+
+    def get_labels(self, filenames: List[str]) -> np.ndarray:
+        labels = np.array([self.dataset[filename][1] for filename in filenames])
+        return labels
+
+
+# %%
+# Definition of dataset, active learning agent and classifier
+dataset = CSVEmbeddingDataset(path_to_embeddings_csv=path_to_embeddings_csv)
+agent = ActiveLearningAgent(ApiWorkflowClient(host=host, token=token, dataset_id=dataset_id))
+classifier = KNeighborsClassifier(n_neighbors=20, weights='distance')
+
+# %%
+# Upload of the embeddings
+agent.api_workflow_client.upload_embeddings(name="embedding-1", path_to_embeddings_csv=path_to_embeddings_csv)
+
+# %%
+# 1. Choose an initial subset of your dataset.
+print("Starting the initial sampling")
+sampler_config = SamplerConfig(name="initial-sampling", n_samples=n_samples_to_sample[0], method=methods_to_sample[0])
+agent.query(sampler_config=sampler_config)
```

Comment:
Same quesiton here as above. What's the reason I have to create the sampler config separately? Why not just call 
`agent.query(name="initial-sampling", n_samples=n_samples_to_sample[0], method=methods_to_sample[0])`

Reply:
As discussed, we keep it like this.